Src/AutoFixtureUnitTest/DataAnnotations/MinAndMaxLengthAttributeRelayTest.cs

```diff
@@ -0,0 +1,273 @@
+        [InlineData(3, 6)]
+        public void CreateWithFiniteSequenceRequestConstrainedbyMinAndMaxLengthReturnsCorrectResult(int min, int max)
+        {
+            // Arrange
+            var memberType = typeof(string[]);
+            var randomNumberWithinRange = (min + max) / 2;
+
+            var minLengthAttribute = new MinLengthAttribute(min);
+            var maxLengthAttribute = new MaxLengthAttribute(max);
+
+            var request = new FakeMemberInfo(
+                new ProvidedAttribute(minLengthAttribute, true),
+                new ProvidedAttribute(maxLengthAttribute, true));
+
+            var expectedRangedNumberRequest = new RangedNumberRequest(typeof(int), min, max);
+            var expectedFiniteSequenceRequest = new FiniteSequenceRequest(
+                memberType.GetElementType(),
+                randomNumberWithinRange);
+
+            var expectedResult = new List<string>();
+
+            var context = new DelegatingSpecimenContext
+            {
+                OnResolve = r =>
+                {
+                    if (r.Equals(expectedRangedNumberRequest)) return randomNumberWithinRange;
+                    if (r.Equals(expectedFiniteSequenceRequest)) return expectedResult;
+                    return new NoSpecimen();
+                }
+            };
+
+            var sut = new MinAndMaxLengthAttributeRelay(
+                new MockMemberTypeResolver { MemberType = memberType });
+
+            // Act
+            var result = sut.Create(request, context);
+
+            // Assert
+            Assert.Equal(expectedResult, result);
+        }
+
+        [Theory]
+        [InlineData(0, 0, 0)]
+        [InlineData(0, 5, 1)]
+        [InlineData(1, 5, 1)]
+        [InlineData(2, 5, 2)]
+        public void WhenMaxValueIsGreaterThen0MinShouldBeEqualToAtLeast1(int min, int max, int expectedMin)
+        {
+            // Arrange
+            var minLengthAttribute = new MinLengthAttribute(min);
+            var maxLengthAttribute = new MaxLengthAttribute(max);
+
+            var request = new FakeMemberInfo(
+                new ProvidedAttribute(minLengthAttribute, true),
+                new ProvidedAttribute(maxLengthAttribute, true));
+
+            ConstrainedStringRequest actualRequest = null;
+
+            var context = new DelegatingSpecimenContext
+            {
+                OnResolve = r =>
+                {
+                    actualRequest = (ConstrainedStringRequest)r;
+                    return new object();
+                }
+            };
+
+            var sut = new MinAndMaxLengthAttributeRelay(
+                new MockMemberTypeResolver());
+
+            // Act
+            sut.Create(request, context);
+
+            //Assert
+            var expectedRequest = new ConstrainedStringRequest(expectedMin, max);
+
+            Assert.Equal(expectedRequest, actualRequest);
+
+        }
+
+        private class MockMemberTypeResolver : IMemberTypeResolver
```

Comment:
Please create `DelegatingMemberTypeResolver` like [this](https://github.com/AutoFixture/AutoFixture/blob/11066d77bcee17e4175679cfa0e9133957695671/Src/AutoFixtureUnitTest/Kernel/DelegatingSpecimenContext.cs#L6) and return `false` by default. I'd expect that each test would configure it rather than relying on default behavior - for better readability.